types/mui-datatables/index.d.ts

```diff
@@ -159,120 +171,132 @@ export interface MUIDataTableIsRowCheck {
-    customSearchRender?: (
+    customRowRender: (data: any[], dataIndex: number, rowIndex: number) => React.ReactNode;
+    customSearch: (searchQuery: string, currentRow: any[], columns: any[]) => boolean;
+    customSearchRender: (
         searchText: string,
         handleSearch: any,
         hideSearch: any,
         options: any
     ) => React.Component | JSX.Element;
-    customSort?: (data: any[], colIndex: number, order: string) => any[];
-    customToolbar?: () => React.ReactNode;
-    customToolbarSelect?: (
-        selectedRows: {
-            data: Array<{ index: number; dataIndex: number }>;
-            lookup: { [key: number]: boolean };
-        },
+    customSort: (data: any[], colIndex: number, order: string) => any[];
+    customTableBodyFooterRender: (options: { data: any[]; selectableRows: SelectableRows; columns: any[] }) => any;
+    customToolbar: () => React.ReactNode;
+    customToolbarSelect: (
+        selectedRows: { data: Array<{ index: number; dataIndex: number }>; lookup: { [key: number]: boolean } },
         displayData: Array<{ data: any[]; dataIndex: number }>,
         setSelectedRows: (rows: number[]) => void
     ) => React.ReactNode;
-    disableToolbarSelect?: boolean;
-    download?: boolean;
-    downloadOptions?: Partial<{
+    download: boolean;
+    downloadOptions: Partial<{
         filename: string;
         separator: string;
         filterOptions: Partial<{ useDisplayedColumnsOnly: boolean; useDisplayedRowsOnly: boolean }>;
     }>;
-    elevation?: number;
-    expandableRows?: boolean;
-    expandableRowsOnClick?: boolean;
-    filter?: boolean;
-    filterType?: FilterType;
-    fixedHeader?: boolean;
-    fixedHeaderOptions?: {
-        xAxis: boolean;
-        yAxis: boolean;
-    };
-    isRowExpandable?: (dataIndex: number, expandedRows?: MUIDataTableIsRowCheck) => boolean;
-    isRowSelectable?: (dataIndex: number, selectedRows?: MUIDataTableIsRowCheck) => boolean;
-    onCellClick?: (
+    elevation: number;
+    enableNestedDataAccess: string;
+    expandableRows: boolean;
+    expandableRowsHeader: boolean;
+    expandableRowsOnClick: boolean;
+    filter: boolean;
+    filterType: FilterType;
+    fixedHeader: boolean;
+    fixedSelectColumn: boolean;
+    isRowExpandable: (dataIndex: number, expandedRows?: MUIDataTableIsRowCheck) => boolean;
+    isRowSelectable: (dataIndex: number, selectedRows?: MUIDataTableIsRowCheck) => boolean;
+    onCellClick: (
         colData: any,
         cellMeta: { colIndex: number; rowIndex: number; dataIndex: number; event: React.MouseEvent }
     ) => void;
-    onChangePage?: (currentPage: number) => void;
-    onChangeRowsPerPage?: (numberOfRows: number) => void;
-    onColumnSortChange?: (changedColumn: string, direction: string) => void;
-    onColumnViewChange?: (changedColumn: string, action: string) => void;
+    onChangePage: (currentPage: number) => void;
```

Comment:
should stay optional IMO:
https://github.com/gregnb/mui-datatables/blob/3ae8302d4c006ee47f951ac0ba113b7fb1b0eba2/src/MUIDataTable.js#L1202
```suggestion
    onChangePage?: (currentPage: number) => void;
```